src/Shared/PackageManagers/PackageManagerFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CST.OpenSource.Shared
+{
+    public class ProjectManagerFactory
+    {
+        // do reflection only once
+        static List<Type> projectManagers = new List<Type>();
+
+        /// <summary>
+        /// Get the project manager for the package type
+        /// </summary>
+        /// <param name="purl"></param>
+        /// <returns>BaseProjectManager object</returns>
+        public static BaseProjectManager CreateProjectManager(PackageURL purl, string destinationDirectory)
```

Comment:
Why use a factory pattern instead of a constructor here?

Factories aren't idiomatic in C#.

Reply:
Since we are following the factory pattern here, it would be better to follow it fully for maintenance purposes, since it is a pretty common pattern used nowadays.